File: C/1000~1999/1110.c
#include <stdio.h>

int main() {
    int n, new_n, cycle, n10, n1;
    scanf("%d", &n);
    cycle = 0;
    new_n = n;
    do {
        cycle++;
        n10 = new_n / 10;
        n1 = new_n % 10;
        new_n = 10 * n1 + (n10 + n1) % 10;
    } while (new_n != n);
    printf("%d", cycle);
}

File: Python/1000~1999/1110.py
def cycle(num):
    return 10*(num%10)+(num//10+num%10)%10

n = int(input())
length = 1
n1 = n

while True:
    n = cycle(n)
    if n1 == n:
        break
    else:
        length += 1
print(length)